src/step-function.ts

```diff
@@ -297,40 +249,52 @@ export namespace $SFN {
    * ```
    * @param paths
    */
-  // @ts-ignore
-  export function parallel<Paths extends readonly (() => any)[]>(
-    ...paths: Paths
-  ): {
-    [i in keyof Paths]: i extends `${number}`
-      ? ReturnType<Extract<Paths[i], () => any>>
-      : Paths[i];
-  };
-
-  export function parallel(call: CallExpr, context: CallContext): ParallelTask {
-    if (context.kind !== ASL.ContextName) {
-      throw new Error(`$SFN.${kind} is only supported by ${ASL.ContextName}`);
-    }
-    const paths = call.getArgument("paths")?.expr;
-    if (paths === undefined) {
-      throw new Error("missing required argument 'paths'");
+  export const parallel = makeIntegration<
+    <Paths extends readonly (() => any)[]>(
+      ...paths: Paths
+    ) => {
+      [i in keyof Paths]: i extends `${number}`
+        ? ReturnType<Extract<Paths[i], () => any>>
+        : Paths[i];
     }
-    if (paths.kind !== "ArrayLiteralExpr") {
-      throw new Error(`invalid arguments to $SFN.parallel`);
-    }
-    ensureItemOf(
-      paths.items,
-      isFunctionExpr,
-      `each parallel path must be an inline FunctionExpr`
-    );
+  >({
+    ...tableIntegrationBase("parallel"),
+    asl(call, context) {
+      const paths = call.getArgument("paths")?.expr;
+      if (paths === undefined) {
+        throw new Error("missing required argument 'paths'");
+      }
+      if (paths.kind !== "ArrayLiteralExpr") {
+        throw new Error(`invalid arguments to $SFN.parallel`);
+      }
+      ensureItemOf(
+        paths.items,
+        isFunctionExpr,
+        `each parallel path must be an inline FunctionExpr`
+      );
 
-    return {
-      Type: "Parallel",
-      Branches: paths.items.map((func) => ({
-        StartAt: context.getStateName(func.body.step()!),
-        States: context.execute(func.body),
-      })),
-    };
-  }
+      return {
+        Type: "Parallel",
+        Branches: paths.items.map((func) => ({
+          StartAt: context.getStateName(func.body.step()!),
+          States: context.execute(func.body),
+        })),
+      };
+    },
+  });
+}
+
+function tableIntegrationBase(
```

Comment:
Why is this called table?

Reply:
It has things that are `Table.operation` specific.